src/main/java/com/gregtechceu/gtceu/api/pattern/BlockPattern.java

```diff
@@ -302,16 +308,15 @@ public void autoBuild(Player player, MultiblockState worldState) {
 
                             // check inventory
                             ItemStack found = null;
-                            ItemStack originalItemStack = null;
+                            int foundSlot = -1;
+                            IItemHandler handler = null;
                             if (!player.isCreative()) {
-                                for (ItemStack itemStack : player.getInventory().items) {
-                                    if (candidates.stream().anyMatch(
-                                            candidate -> ItemStack.isSameItemSameTags(candidate, itemStack)) &&
-                                            !itemStack.isEmpty() && itemStack.getItem() instanceof BlockItem) {
-                                        found = itemStack.copy();
-                                        originalItemStack = itemStack;
-                                        break;
-                                    }
+                                var founded = getMatchStackWithHandler(candidates,
```

Comment:
name the field `foundHandler` instead of `founded`

Reply:
renamed